.github/workflows/build.yml

```diff
@@ -0,0 +1,327 @@
+name: build
+
+on:
+  push:
+    branches: [ master, main ]
+    tags: [ 'v*' ]
+  pull_request:
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - run: sudo bin/build-linux.sh
+
+    - run: |
+        ./rubyc-linux-x64 -v
+        ./rubyc-linux-x64 -V
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: rubyc-linux-x64
+        path: rubyc-linux-x64
+
+  macos:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ruby/setup-ruby@v1
+
+    - uses: shogo82148/actions-setup-perl@v1
+      with:
+        perl-version: '5.16.2'
+
+    - run: brew install squashfs
+
+    - run: bin/build-macos.sh
+
+    - run: |
+        ./rubyc-darwin-x64 -v
+        ./rubyc-darwin-x64 -V
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: rubyc-darwin-x64
+        path: rubyc-darwin-x64
+
+  linux-packed-mn:
+    if: ${{ !startsWith(github.ref, 'refs/tags/') }}
+    needs: linux
+    runs-on: ubuntu-16.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        repository: metanorma/packed-mn
+
+    - uses: ruby/setup-ruby@v1
+      with:
+        bundler: 2.2.3
+        bundler-cache: true
+
+    - run: bundle update --all
+
+    - run: sudo apt-get update && sudo apt-get -y -q install squashfs-tools bison flex texinfo
+
+    - run: |
+        sudo update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100
+        sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
+
+    - run: |
+        curl -Lks 'https://git.io/rg-ssl' | ruby
+        echo "perl -v: $(perl -v)"
+        echo "clang -v: $(clang -v)"
+        echo "clang++ -v: $(clang -v)"
+
+    - name: enable swap
+      run: |
+        sudo fallocate -l 9g /mnt/compile.swap
+        sudo chmod 600 /mnt/compile.swap
+        sudo mkswap /mnt/compile.swap
+        sudo swapon /mnt/compile.swap
+        sudo swapon --all --verbose
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: rubyc-linux-x64
+
+    - run: mv rubyc-linux-x64 rubyc && chmod +x rubyc
+
+    - run: make build/metanorma
+      timeout-minutes: 180
+      # https://trac.sagemath.org/ticket/19762
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+
+    - run: build/metanorma help
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: metanorma-linux
+        path: build/metanorma
+
+  linux-test:
+    needs: linux-packed-mn
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.ignore-errors }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-16.04, ubuntu-18.04, ubuntu-20.04 ]
+        ignore-errors: [ false ]
+        test-flavor:
+          - iso
```

Comment:
I propose to reduce this list just to iso

Reply:
Agree.